File: mvvmhabit/src/main/java/me/goldze/mvvmhabit/binding/viewadapter/tablayout/ViewAdapter.java
package me.goldze.mvvmhabit.binding.viewadapter.tablayout;

import androidx.databinding.BindingAdapter;

import com.google.android.material.tabs.TabLayout;

import me.goldze.mvvmhabit.binding.command.BindingCommand;

/**
 * @author Alwyn
 * @Date 2020/10/31
 * @Description
 */
public class ViewAdapter {
    @BindingAdapter("onTabSelectedCommand")
    public static void onTabSelectedCommand(TabLayout tabLayout, final BindingCommand<Integer> bindingCommand) {
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                if (bindingCommand != null) {
                    bindingCommand.execute(tab.getPosition());
                }
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
    }
}

File: lib_base/src/main/java/com/czl/lib_base/binding/viewadapter/checkbox/ViewAdapter.kt
package com.czl.lib_base.binding.viewadapter.checkbox

import android.widget.CheckBox
import androidx.databinding.BindingAdapter
import com.czl.lib_base.binding.command.BindingCommand

object ViewAdapter {
    /**
     * @param bindingCommand //绑定监听
     */
    @JvmStatic
    @BindingAdapter(value = ["onCheckedChangedCommand"], requireAll = false)
    fun setCheckedChanged(checkBox: CheckBox, bindingCommand: BindingCommand<Boolean?>) {
        checkBox.setOnCheckedChangeListener { compoundButton, b -> bindingCommand.execute(b) }
    }
}